.github/workflows/ci.yml

```diff
@@ -23,7 +23,6 @@ jobs:
         pip install -r requirements.txt
     - name: Check
       run: |
-        pre-commit run --all-files
         make check
```

Comment:
Since we use the pre-commit CI, we can simplify this here. We could remove the entire python setup as well as clang-format. We only need to install `shfmt` and then call `make check`

Reply:
Done. Let's try to migrate shfmt to pre-commit in a follow-up PR. I see plenty of shfmt hooks at https://pre-commit.com/hooks.html, might be possible.